### PR TITLE
Add support for OPT_PASSTHROUGH_ENUM

### PR DESCRIPTION
**Pull request by FerasAlazzeh**
Passthrough `enum.Enum` instances to `default`. This allows serializing Enums to a custom format.

### CODE REVIEW

**Comment by FerasAlazzeh** [comment]
A use case for this is to help people migrate from other libs to `orjson`. Take for example:

```
>>> import orjson
>>> import simplejson
>>> import enum
>>> names = enum.Enum('names', ('alice', 'bob'))
b'1'
>>> simplejson.dumps(names.alice)
TypeError: Object of type names is not JSON serializable

>>> json.dumps(names.alice)
TypeError: Object of type names is not JSON serializable

This allows users to specify a custom format for enums if they have already specified a custom format using another lib.
```

**Comment by ijl** [comment]
I think the concept is reasonable, but this needs documentation. In particular, the interaction with subclases needs to be obvious. Please generally sort things and use the `lint` script as well.

File: test/test_enum.py [hunk 1]
```diff
@@ -116,3 +116,47 @@ def test_non_str_keys_enum(self):
...
+        IntFlagEnum is a subclass of int so should be unaffected by the passthrough flag
+        """
+        self.assertEqual(orjson.dumps(IntFlagEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM), b"1")
+
+    def test_passthrough_default_enum_default(self):
+        def default(obj):
+            return str(obj)
+
+        names = enum.Enum("names", ("john", "alice"))
+        self.assertEqual(orjson.dumps(names.alice, option=orjson.OPT_PASSTHROUGH_ENUM, default=default), b'"names.alice"')
```
Comment (ijl) [inline test/test_enum.py:161]: Should "names.alice" be the default behavior? I'm not familiar.

**Comment by FerasAlazzeh** [comment]
Sounds good ran the `./lint` script and re-pushed.

**Comment by ijl** [comment]
Ok, could you document it as well?

**Comment by FerasAlazzeh** [comment]
Essentially any Enum that is not `int, float, int- & float-derived Enums` can be passed through.

Meaning the following Enum Classes cannot be passed through:
`enum.IntFlag` - Base class for creating enumerated constants that are also subclasses of int.
`enum.IntEnum` -  Base class for creating enumerated constants that are also subclasses of int.
`enum.IntFlag` - Base class for creating enumerated constants that can be combined using the bitwise operators without losing their IntFlag membership. IntFlag members are also subclasses of int.

Enum types that can be passed through are:
`enum.Enum`
`enum.Flag`

**Comment by FerasAlazzeh** [comment]
Sounds good I will document this as well.

File: test/test_enum.py (hunk 1 above)
Comment (FerasAlazzeh) [reply test/test_enum.py]: I do not think so. I think how its implemented is fine. 

If you look at https://docs.python.org/3/library/json.html#json.JSONEncoder
You will see

```
Python | Json
...
int, float, int- & float-derived Enums | number
```

So the default serialization is correct.

**Comment by FerasAlazzeh** [comment]
Just updated running the `./lint` and also adding documented explicitly the usage and which classes this would apply too.

**Comment by FerasAlazzeh** [comment]
bump waiting for a review :)

**Comment by ijl** [comment]
I don't follow from the documentation why certain types are passed through and others aren't. There's a rule but it's not explained. Can you add that? The list won't be kept up to date with any changes in the standard library, so the rule is the important thing. I think the non-obvious case that subclasses will be passed through even with this flag needs its own clear, don't shoot your foot off example. It also needs tests (any assetion in the documentation is a test). Thsi is minor but OPT_PASSTHROUGH_ENUM and whatever should be kept in alphabetical order as well.

**Comment by ijl** [comment]
Feel free to reopen.

**Pull request closed without merging**